concent_api/core/payments/report.py

```diff
@@ -0,0 +1,759 @@
+        min_block_timestamp=get_oldest_subtask_creation_date(),
+        transaction_type=TransactionType.FORCE,
+    )
+
+    # Additional verification payments.
+    additional_verification_payments_for_requestor = service.get_covered_additional_verification_costs(  # pylint: disable=no-value-for-parameter
+        client_eth_address=requestor_ethereum_address,
+        payment_ts=get_oldest_subtask_creation_date(),
+    )
+    additional_verification_payments_for_provider = service.get_covered_additional_verification_costs(  # pylint: disable=no-value-for-parameter
+        client_eth_address=provider_ethereum_address,
+        payment_ts=get_oldest_subtask_creation_date(),
+    )
+
+    return Payments(
+        regular_payments=regular_payments,
+        settlement_payments=settlement_payments,
+        forced_subtask_payments=forced_subtask_payments,
+        additional_verification_payments_for_requestor=additional_verification_payments_for_requestor,
+        additional_verification_payments_for_provider=additional_verification_payments_for_provider,
+    )
+
+
+def find_forced_subtask_payments_for_subtask_results_settled(
+    subtask_results_settled: PendingResponse,
+    forced_subtask_payments: list,
+) -> List[ForcedSubtaskPaymentEvent]:
+    """ Find forced subtask payments for the same subtask for given SubtaskResultsSettled message. """
+    return [
+        forced_subtask_payment for forced_subtask_payment in forced_subtask_payments
+        if forced_subtask_payment.subtask_id == subtask_results_settled.subtask.subtask_id
+    ]
+
+
+def find_subtask_result_for_additional_verification_payment(
+    additional_verification_payment: CoverAdditionalVerificationEvent,
+    subtask_results_settled: list,
+    subtask_results_rejected: list,
+) -> List[PendingResponse]:
+    """ Find the SubtaskResultsSettled or SubtaskResultsRejected for given additional verification payment. """
+    matched_subtask_results_settled = [
+        results_settled for results_settled in subtask_results_settled
+        if results_settled.subtask.subtask_id == additional_verification_payment.subtask_id
+    ]
+
+    matched_subtask_results_rejected = [
+        results_rejected for results_rejected in subtask_results_rejected
+        if results_rejected.subtask.subtask_id == additional_verification_payment.subtask_id
+    ]
+
+    return matched_subtask_results_settled + matched_subtask_results_rejected
+
+
+def find_settlement_payment_for_force_payment_committed(
+    force_payment_committed_pending_response: PendingResponse,
+    settlement_payments: list,
+) -> List[ForcedPaymentEvent]:
+    """ Find settlement payment for given ForcePaymentCommitted message. """
+    payment_info = force_payment_committed_pending_response.payments.order_by('id').last()
+    return [
+        settlement_payment for settlement_payment in settlement_payments
+        if (
+            int(parse_datetime_to_timestamp(payment_info.payment_ts)) - 10 <=
```

Comment:
`payment_ts` is parsed from datetime to timestamp? It should be timestamp  without parsing.

Reply:
Yes, exactly as @cameel wrote, it is a database column which casted to `datetime` object.